client/webserver/http.go

```diff
@@ -190,6 +197,109 @@ func (s *WebServer) handleOrders(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+// handleExportOrders is the handler for the /orders/export page request.
+func (s *WebServer) handleExportOrders(w http.ResponseWriter, r *http.Request) {
+	filter := new(core.OrderFilter)
+	err := r.ParseForm()
+	if err != nil {
+		log.Errorf("error parsing form for export order: %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	filter.Hosts = r.Form["hosts"]
+	assets := r.Form["assets"]
+	filter.Assets = make([]uint32, len(assets))
+	for k, assetStrId := range assets {
+		assetNumId, err := strconv.Atoi(assetStrId)
+		if err != nil {
+			log.Errorf("error parsing asset id: %v", err)
+			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			return
+		}
+		filter.Assets[k] = uint32(assetNumId)
+	}
+	statuses := r.Form["statuses"]
+	filter.Statuses = make([]order.OrderStatus, len(statuses))
+	for k, statusStrId := range statuses {
+		statusNumId, err := strconv.Atoi(statusStrId)
+		if err != nil {
+			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			log.Errorf("error parsing status id: %v", err)
+			return
+		}
+		filter.Statuses[k] = order.OrderStatus(statusNumId)
+	}
+
+	ords, err := s.core.Orders(filter)
+	if err != nil {
+		log.Errorf("error retrieving order: %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	var buffer bytes.Buffer
+	csvWriter := csv.NewWriter(&buffer)
+	// UseCRLF for Window users
+	csvWriter.UseCRLF = strings.Contains(r.UserAgent(), "Windows")
+
+	err = csvWriter.Write([]string{
+		"Host",
+		"Base",
+		"Quote",
+		"Ask",
+		"Offer",
+		"Type",
+		"Status",
+		"Rate",
+		"Filled (%)",
+		"Settled (%)",
+		"Quantity",
+		"Time",
+	})
+	if err != nil {
+		log.Errorf("error writing CSV: %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+	for _, ord := range ords {
+		ordReader := orderReader{ord}
+
+		srvTime := encode.UnixTimeMilli(int64(ord.Stamp))
+		err = csvWriter.Write([]string{
+			ord.Host,                                  // Host
+			ord.BaseSymbol,                            // Base
+			ord.QuoteSymbol,                           // Quote
+			ordReader.AskString(),                     // Ask
+			ordReader.OfferString(),                   // Offer
+			ordReader.TypeString(),                    // Type
+			ordReader.StatusString(),                  // Status
+			ordReader.RateString(),                    // Rate
+			ordReader.FilledPercent(),                 // Filled
+			ordReader.SettledPercent(),                // Settled
+			ordReader.OfferString(),                   // Quantity
+			srvTime.Format("02/01/2006, 03:04:05 PM"), // Time
+		})
+		if err != nil {
+			log.Errorf("error writing CSV: %v", err)
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			return
+		}
+	}
+	csvWriter.Flush()
```

Comment:
Poking in the [`encoding/csv.Writer docs`](https://golang.org/pkg/encoding/csv/#Writer.Error), it looks like we're supposed to call `csvWriter.Error()` after `Flush`:

> After all data has been written, the client should call the Flush method to guarantee all data has been forwarded to the underlying io.Writer. Any errors that occurred should be checked by calling the Error method.

> Flush writes any buffered data to the underlying io.Writer. To check if an error occurred during the Flush, call Error.